grpc-sys/src/lib.rs

```diff
@@ -379,24 +379,184 @@ pub union GrpcSliceData {
 pub enum GrpcSliceRefCount {}
 
 #[repr(C)]
-#[derive(Clone, Copy)]
-pub struct GrpcSlice {
+pub struct GrpcSliceRaw {
     ref_count: *mut GrpcSliceRefCount,
     data: GrpcSliceData,
 }
 
+/// Trivial copy. We need to copy the inlined-array, which means we can't
+/// Simply `#[derive(Clone)]`.
+/// Here we use the API provided by gRPC.
+impl Clone for GrpcSliceRaw {
+    fn clone(&self) -> Self {
+        unsafe { grpcwrap_slice_copy(self as _) }
+    }
+}
+
+impl<'a> From<&'a [i8]> for GrpcSliceRaw {
```

Comment:
Why is this impl added 🤔.

Reply:
Dunno. Will delete it later

Reply:
Can't delete, required in the test

Reply:
That's odd, which test produces an `&[i8]` (or `&[c_char]`)? This normally shouldn't happen in Rust

Reply:
Sorry, deleting it works now.